frontend/src/screens/ShippingAddressScreen.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from "react";
+import { Form, Button } from "react-bootstrap";
+import { useDispatch, useSelector } from "react-redux";
+import FormContainer from "../components/FormContainer";
+import { saveShippingAddress } from "../services/cartAction";
+import store from "../store";
+
+const ShippingAddressScreen = ({ history }) => {
+  //
+  // pull put data from Store (state.cart)
+  const cart = useSelector((state) => state.cart);
+  const { shippingAddress } = cart;
+
+  //state for data updated by form onChange actions
+  const [name, setName] = useState(shippingAddress.name);
```

Comment:
"Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. "

If the shippingAddress info already exists in the state from a previous form submission, then the states are initialized to those values. If the state does not yet exist, then those values are undefined, and the warning occurs when the input is changed because it's going from undefined to defined. This can be fixed by first checking if the values are set, and if they're not, then you can do something like 
const [name, setName] = useState('');
etc.

Reply:
Agreed in weekly meeting that fixing this is not a priority. It's only a warning and not an error, only displays in the console so users don't see it, and it doesn't affect the functionality. Additionally, it was not reproducible for all members; those with Macs did not experience this warning.